src/command/subgraph/check.rs

```diff
@@ -0,0 +1,140 @@
+use anyhow::{Context, Result};
+use prettytable::{cell, row, Table};
+use serde::Serialize;
+use structopt::StructOpt;
+
+use rover_client::query::subgraph::check;
+
+use crate::client::StudioClientConfig;
+use crate::command::RoverStdout;
+use crate::utils::loaders::load_schema_from_flag;
+use crate::utils::parsers::{parse_graph_ref, parse_schema_source, GraphRef, SchemaSource};
+
+#[derive(Debug, Serialize, StructOpt)]
+pub struct Check {
+    /// <NAME>@<VARIANT> of graph in Apollo Studio to validate.
+    /// @<VARIANT> may be left off, defaulting to @current
+    #[structopt(name = "GRAPH_REF", parse(try_from_str = parse_graph_ref))]
+    #[serde(skip_serializing)]
+    graph: GraphRef,
+
+    /// Name of the implementing service to validate
+    #[structopt(long = "service", required = true)]
+    #[serde(skip_serializing)]
+    service_name: String,
+
+    /// Name of configuration profile to use
+    #[structopt(long = "profile", default_value = "default")]
+    #[serde(skip_serializing)]
+    profile_name: String,
+
+    /// The schema file to push
+    /// Can pass `-` to use stdin instead of a file
+    #[structopt(long, short = "s", parse(try_from_str = parse_schema_source))]
+    #[serde(skip_serializing)]
+    schema: SchemaSource,
+}
+
+impl Check {
+    pub fn run(&self, client_config: StudioClientConfig) -> Result<RoverStdout> {
+        let client = client_config.get_client(&self.profile_name)?;
+
+        let sdl = load_schema_from_flag(&self.schema, std::io::stdin())?;
+
+        let partial_schema = check::check_partial_schema_query::PartialSchemaInput {
+            sdl: Some(sdl),
+            // we never need to send the hash since the back end computes it from SDL
+            hash: None,
+        };
+        let res = check::run(
+            check::check_partial_schema_query::Variables {
+                graph_id: self.graph.name.clone(),
+                variant: self.graph.variant.clone(),
+                partial_schema,
+                implementing_service_name: self.service_name.clone(),
+            },
+            &client,
+        )
+        .context("Failed to validate schema")?;
+
+        tracing::info!(
+            "Validated the proposed subgraph against metrics from {}@{}",
```

Comment:
I don't like the way this sounds. `validated` makes it sound like it was successful, and `metrics` implies there is usage metrics reported for a service (which may or may not be true).
```suggestion
            "Checked the proposed subgraph against {}@{}",
```